src/solver/qualrout.c

```diff
@@ -330,8 +364,22 @@ void findLinkQual(int i, double tStep)
             c2 = 0.0;
         }
 
-        // --- assign new concen. to link
-        Link[i].newQual[p] = c2;
+	if (Link[i].extPollutFlag[p] == 0)
+	{
+            // --- assign new concen. to link
+            Link[i].newQual[p] = c2;
+	}
+	// --- update mass balance and set external pollutant
+	else if( Link[i].extPollutFlag[p] == 1)
+	{
+	    // --- mass balance update
+            lossExtQual = c2 - Link[i].extQual[p];
+            lossExtQual = lossExtQual * v1/ tStep;
+            massbal_addReactedMass(p, lossExtQual);
+
+            Link[i].newQual[p] = Link[i].extQual[p];
+            Link[i].extPollutFlag[p] = 0;
+	}
```

Comment:
Conceptually, having difficulty understanding the necessity of associating an external pollutant with a link. How would a pollutant enter a link?

Reply:
I get it now. This is for modeling an eroding channel. Or a channel that has accumulated a large amount of sediment that is getting resuspended during storm events.